src/ItemDetail.js

```diff
@@ -12,18 +12,35 @@ import {
 
 import { openRequestStatusFilters } from './utils';
 
-const ItemDetail = ({ item, loan, requestCount }) => {
-  if (!item.id && !item.barcode) {
+const MISSING_VALUE_SYMBOL = '-';
+
+const ItemDetail = ({
+  currentInstanceId,
+  request,
+  item,
+  loan,
+  requestCount,
+}) => {
+  const itemId = request?.itemId || item.id;
+
+  if (!itemId && !item.barcode) {
     return <FormattedMessage id="ui-requests.actions.loading" />;
   }
 
-  const recordLink = item ? <Link to={`/inventory/view/${item.instanceId}/${item.holdingsRecordId}/${item.id}`}>{item.barcode || item.id}</Link> : '-';
-  const status = get(item, 'status.name') || get(item, 'status');
-  const contributor = get(item, ['contributorNames', '0', 'name'], '-');
+  const instanceId = request?.instanceId || currentInstanceId;
+  const holdingsRecordId = request?.holdingsRecordId || item.holdingsRecordId;
+  const title = request?.instance.title || item.title || MISSING_VALUE_SYMBOL;
+  const contributor = get(request, ['instance', 'contributorNames', '0', 'name']) || get(item, ['contributorNames', '0', 'name']) || MISSING_VALUE_SYMBOL;
```

Comment:
@Stanislau-Snapok this would be a good opportunity to get rid of `get` and replace it with `?`

Reply:
Yeah, you right! Changed, thanks!